apps/aechannel/src/aesc_fsm.erl

```diff
@@ -4011,3 +4270,37 @@ block_hash_from_op(?NO_OP) -> %% in case of unexpected close
 
 tx_env_and_trees_from_top(Type) ->
     aesc_tx_env_cache:tx_env_and_trees_from_top(Type).
+
+-spec check_block_hash(aec_keys:pubkey(), #data{}) -> ok.
+check_block_hash(?NOT_SET_BLOCK_HASH, _) -> ok;
+check_block_hash(BlockHash,
+                 #data{block_hash_delta = #bh_delta{ not_older_than = LowerDelta
+                                                   , not_newer_than = UpperDelta}}) ->
+    case aec_chain:get_header(BlockHash) of
+        {ok, Header} ->
+            Checks =
+                [ fun() ->
+                      CurrHeight = curr_height(),
+                      BlockHeight = aec_headers:height(Header),
+                      UpperLimit = CurrHeight - UpperDelta,
+                      LowerLimit = CurrHeight - LowerDelta,
+                      case BlockHeight of
+                          _ when BlockHeight > UpperLimit ->
+                              {error, block_hash_too_new};
```

Comment:
There is currently no way for a client to know how new or old the block hash was. This leads to a situtation of lots of trial and error to get into the expected range. Would it make sense to add the difference to the expected range into the error response?

Reply:
I was concerned that exposing one's range could open the door for some poking in order for a malicious party to know the node's view of the chain and how well connected it is (ex: when what top it receives)